src/app/common/theme-provider.tsx

```diff
@@ -27,12 +30,24 @@ const ThemeContext = createContext<{
 const getSystemTheme = () =>
   window.matchMedia('prefers-color-scheme: dark').matches ? 'dark' : 'light';
 
+function getComputedTheme(userSelectedTheme: UserSelectedTheme): ComputedTheme {
+  if (userSelectedTheme === 'system') {
+    return getSystemTheme();
+  }
+
+  return userSelectedTheme;
+}
+
+function setUserSelectedTheme(theme: UserSelectedTheme) {
+  store.dispatch(userSelectedThemeActions.setUserSelectedTheme(theme));
+}
+
 interface ThemeSwitcherProviderProps {
   children: JSX.Element | JSX.Element[];
 }
 export const ThemeSwitcherProvider = ({ children }: ThemeSwitcherProviderProps) => {
-  const [userSelectedTheme, setUserSelectedTheme] = useState<UserSelectedTheme>('system');
-  const [theme, setTheme] = useState<ComputedTheme>(() => getSystemTheme());
+  const userSelectedTheme = useUserSelectedThemeValue();
+  const [theme, setTheme] = useState<ComputedTheme>(() => getComputedTheme(userSelectedTheme));
```

Comment:
Is there a reason to introduce local state too rather than just relying on redux?

Reply:
The reasoning behind this extra layer of state is that this is the actual theme (light or dark) being applied, which is computed based off of the "user selected theme" (which additionally includes `system`). We only care about storing what the user selected, not what the app computed based off of the user's selection. The computed theme depends on the user's selection and, for "system", their system settings.

Reply:
Got it, thx for clarifying. 👍